### ISSUE TITLE
Postgres process sometime listen on localhost instead of 0.0.0.0

### ISSUE THREAD

**Issue by pckhoi** [docker-library/postgres#583]
I run a custom postgres image with docker-compose and sometime I cannot connect to it, and it is always because Postgres process somehow bound to `127.0.0.1:5432` instead of `0.0.0.0:5432`. It is driving me nut because sometime it bind correctly to `0.0.0.0` sometime it bind to `127.0.0.1` instead:
```
postgres_1    | 2019-05-22 03:20:13.657 GMT [1] LOG:  listening on IPv4 address "127.0.0.1", port 5432
postgres_1    | 2019-05-22 03:20:13.662 GMT [1] LOG:  could not bind IPv6 address "::1": Address not available
postgres_1    | 2019-05-22 03:20:13.662 GMT [1] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
```

Below is my docker-compose config:

```
  postgres:
    build:
      context: ./postgres
    ports:
      - 5432:5432
    volumes:
      - pgdata:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=wrgl
      - POSTGRES_PASSWORD=password
```

My custom postgres image only has an additional script to change a couple lines in postgresql.conf.

Docker context:
```
postgres
    |_  Dockerfile
    |_  enable_log_statement.sh
```
Dockerfile:
```
FROM postgres:11.3-alpine

ADD . /docker-entrypoint-initdb.d
```
enable_log_statement.sh:
```
#!/bin/sh
set -x

PGCONF=/var/lib/postgresql/data/postgresql.conf
cat $PGCONF | sed "s/#log_statement = 'none'.*/log_statement = 'all'/g" > $PGCONF
cat $PGCONF | sed "s/#listen_addresses = 'localhost'.*/listen_addresses = '*'/g" > $PGCONF
```

**Comment by wglambert** [comment]
Your script is creating a blank postgresql.conf, `sed` doesn't like an output redirection to the same file it's editing, it has a `-i` switch for in-place editing.

Also the listen_address in the image's postgresql.conf is already `*`. And sed accepts the last argument as the input-file

So change your script to `sed -i "s/#log_statement = 'none'.*/log_statement = 'all'/g" $PGCONF`

**Comment by pckhoi** [comment]
Hello, thanks for the suggestion. I looked at the file's content right after so I'm pretty sure those lines worked. But nevertheless I changed the script like you suggested and rather mysteriously it's now working again. But please keep this issue open for a while. I suspect it may come back.

**Comment by sc0v1n0** [comment]
This question is now closed, but if I can leave a comment in case someone has a similar problem in the future, I had the same problem above that started. Using docker compose and SQLAlchemy, the problem was not in SQLAlchemy, looking for Docker compose, my teammate had this observation of the ports, to solve the problem we had to reverse the ports. After that we had no more problems, the access through SQLAlchemy went smoothly.

![image](https://user-images.githubusercontent.com/56802572/193296050-a8bb7fca-e742-42f9-9937-0f4c737f4002.png)

![image](https://user-images.githubusercontent.com/56802572/193297153-a1667710-5e2f-45bb-893c-e9cc28961a9d.png)
